src/plugins/matrixops/eye_operation.cpp

```diff
@@ -0,0 +1,232 @@
+    {
+        if (extract_numeric_value_dimension(arg) != 0)
+        {
+            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                "eye_operation::eye_n_helper",
+                generate_error_message("input should be a scalar"));
+        }
+        std::int64_t N = arg.scalar();
+        if (N < 0)
+        {
+            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                "eye_operation::eye_n_helper",
+                generate_error_message("input should be greater than zero"));
+        }
+        std::size_t size = static_cast<std::size_t>(N);
+        return primitive_argument_type{
+            ir::node_data<T>{blaze::IdentityMatrix<T>(size)}};
+    }
+
+    primitive_argument_type eye_operation::eye_n(arg_type&& arg) const
+    {
+        node_data_type t = dtype_;
+
+        switch (t)
+        {
+        case node_data_type_bool:
+            return eye_n_helper<std::uint8_t>(std::move(arg));
+
+        case node_data_type_int64:
+            return eye_n_helper<std::int64_t>(std::move(arg));
+
+        case node_data_type_unknown:
+            HPX_FALLTHROUGH;
+        case node_data_type_double:
+            return eye_n_helper<double>(std::move(arg));
+
+        default:
+            break;
+        }
+
+        HPX_THROW_EXCEPTION(hpx::bad_parameter,
+            "phylanx::execution_tree::primitives::"
+            "eye_operation::eye_n",
+            generate_error_message(
+                "the eye primitive requires for all arguments to "
+                "be numeric data types"));
+    }
+
+    template <typename T>
+    primitive_argument_type eye_operation::eye_nmk_helper(args_type&& args) const
+    {
+        if (extract_numeric_value_dimension(args[0]) != 0 ||
+            extract_numeric_value_dimension(args[1]) != 0)
+        {
+            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                "eye_operation::eye_n_helper",
+                generate_error_message("input should be a scalar"));
+        }
+        auto N = args[0].scalar();
+        auto M = args[1].scalar();
+        auto k = 0;
+        if (args.size() == 3)
+        {
+            if (extract_numeric_value_dimension(args[2]) != 0)
+            {
+                HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                    "eye_operation::eye_n_helper",
+                    generate_error_message("input should be a scalar"));
+            }
+            k = args[2].scalar();
+        }
+
+        if (N < 0 || M < 0)
+        {
+            HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                "eye_operation::eye_nm_helper",
+                generate_error_message("inputs should be greater than zero"));
+        }
+        blaze::DynamicMatrix<T> result(N, M, T(0));
+
+        auto vecsize = static_cast<T>(1);    // a positive number (k=0)
+
+        if (k > 0)
+            vecsize = M - k;
+        else if (k < 0)
+            vecsize = N + k;
+
+        if (vecsize > 0)
+            blaze::band(result, k) = T(1);
```

Comment:
If `k == 0` this should return the identity matrix. AFAICS, currently the code returns a matrix with all zeros.

Reply:
I think when `k == 0` we have `vecsize == 1` and since `vecsize` is positive we still have `blaze::band(result, k) = T(1)`
Please tell me if I am missing something.

Reply:
You're right, I misunderstood the code.